src/widgets/list.rs

```diff
@@ -73,6 +77,64 @@ impl<'a> ListItem<'a> {
     pub fn width(&self) -> usize {
         self.content.width()
     }
+
+    pub fn render(&self, area: Rect, buffer: &mut Buffer) {
+        self.content.render(area, buffer);
+    }
+}
+
+pub struct WidgetWrapper<'a> {
+    widget: Box<dyn Widget + 'a>,
+    height: usize,
+    width: usize,
+}
+
+impl<'a> WidgetWrapper<'a> {
+    pub fn new(widget: impl Widget + 'a, height: usize, width: usize) -> Self {
+        Self {
+            widget: Box::new(widget),
```

Comment:
it is really a nice alternative solution.

1. In my implementation I tried to avoid the heap allocations at all cost. but maybe it is fine to use the heap instead of generic types / templates. it makes the code a little bit cleaner. When I have time, maybe I will try to do some benchmarks with a long list.
2. I like that the render function takes a reference instead of consuming self. This will make some problems easier to solve.